packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -848,17 +939,46 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
         )
       },
       useQuery(arg, options) {
+        const isSkipped: boolean = arg === skipToken || !!options?.skip
         const querySubscriptionResults = useQuerySubscription(arg, options)
         const queryStateResults = useQueryState(arg, {
-          selectFromResult:
-            arg === skipToken || options?.skip
-              ? undefined
-              : noPendingQueryStateSelector,
+          selectFromResult: isSkipped ? undefined : noPendingQueryStateSelector,
           ...options,
         })
+
+        const stableArg = useStableQueryArgs(
+          options?.skip ? skipToken : arg,
+          serializeQueryArgs,
+          context.endpointDefinitions[name],
+          name
+        )
+
+        const prefetch = usePrefetch(name as any)
```

Comment:
using prefetch here might be problematic since that creates a value subscription that cannot be unsubscribed. if we use that here we have to look into that

Reply:
<img width="646" alt="Screenshot 2022-04-18 at 20 35 18" src="https://user-images.githubusercontent.com/16583125/163857176-be3083ba-2980-4e0a-ab93-67a3094a5bc9.png">


You're right :/

 I just need a way to fetch once and then to store  the result.

Reply:
I think at this point we probably need to overwork `prefetch` so that it takes something like a "keepFor" timer to add a time-based temporary subscription (maybe with a fixed subscription id of "prefetch" so that multiple prefetches just "bump up the timer").

I hate to bother you to go deeper into this rabbit hole - but could you look into this?

Reply:
Done https://github.com/reduxjs/redux-toolkit/pull/2245/commits/f9944185dce95c781315e1d86d56a011c68a1fa5 :) 
Please let me know what you think 
I'm still not sure about adding a return type to prefetch but I'll look into it.